Add dark mode to site

lukewarlow opened the pull request.

lukewarlow commented:
These colours have currently mostly been picked at random based on my perception of their contrast so its possible they dont meet requirements (particularly some of the links look slightly dark compared to others so need to check that).

But I think it's a big win having dark mode.

mfreed7 reviewed (approved):
Seems ok to me, but I'm definitely not an expert in this area. I'll click Approve, but it'd be good to get someone more authoritative to review, maybe?

lukewarlow commented:
@gregwhitworth do you have thoughts on the use of modern styling such as light-dark() Vs using preference media queries (basically what browser support do we care about?)

Also do you have thoughts on this change in general.

gregwhitworth commented:
@lukewarlow first off thanks for tackling this. Looking at a few of the pages we have contrast issues that will have to be addressed to land this. The component matrix:

<img width="849" alt="image" src="https://github.com/user-attachments/assets/84cae961-01e1-492b-aeb4-62bb55146dbe" />

Code blocks. Here is dark mode:

<img width="836" alt="image" src="https://github.com/user-attachments/assets/8e6cc0bd-c33e-46c4-877c-664d3ec50345" />

Versus light mode:

<img width="699" alt="image" src="https://github.com/user-attachments/assets/5797e11d-41fd-4b8a-9b4d-bc2f5aa9abbe" />

The anatomy aren't visible due to leveraging transparency and assuming a white background:

<img width="875" alt="image" src="https://github.com/user-attachments/assets/c85acddd-cc0b-4f14-9733-82bd8a3650b7" />

The [.note classes](https://deploy-preview-1168--open-ui.netlify.app/components/checkbox.research.findings/) likewise end up with poor contrast:

<img width="715" alt="image" src="https://github.com/user-attachments/assets/0959f16f-83e1-4f35-81d4-13ff9d9a7786" />

Same with the Slider ["Parts and Concepts" table](https://deploy-preview-1168--open-ui.netlify.app/components/slider.research.parts/)

<img width="664" alt="image" src="https://github.com/user-attachments/assets/a0790942-b4ca-418f-8a3f-ea2901313eb9" />


Looking at the various High Contrast solutions and or Color Pages they seem fine. So I think if you tweak the above two we should be in a good spot.

gregwhitworth reviewed (changes requested):
See prior comment, a few scenarios end up with poor contrast due to these changes

lukewarlow commented:
I've changed everything that I could find, I don't understand what you mean about anatomies? They seem to render fine (per your screenshot?)

lukewarlow commented:
@gregwhitworth I believe I've addressed all the concerns you've raised, other than the bit about anatomies which I don't understand what you mean? They appear to work correctly to me at least.
